### DISCUSSION TITLE
How do I skip PCA in make_graph?

### DISCUSSION THREAD

**Discussion by rjesud** [parashardhapola/scarf#108] [General]
Hi! This tool is a welcome solution for large single-cell datasets coming from tissue image-based single-cell datasets where datasets can easily grow >10M cells.

In protein imaging datasets we often do not need to run PCA since the plex is relavativelty low (8-40plex). In scanpy I typically skip PCA and jump right into neighborhood creation with the full feature set. Since PCA is integrated with make_graph in Scarf it is less clear to me how I would do this.

Can you provide any guidance here?

**Comment by parashardhapola** [comment]
Hi thank you for the question!

Yes, if you set `dims=0` in `make_graph` then the linear dimension reduction (PCA for RNA-Seq and LSI for ATAC-Seq) step is skipped.

If your data is compositional, then maybe you should treat it as "ADT" (antibody derived tags) which uses CLR-normalization. Check out the related section of the tutorial:
https://scarf.readthedocs.io/en/latest/vignettes/multiple_modalities.html#process-protein-surface-abundance-modality

Best regards,
Parashar
